### ISSUE TITLE
After disconnection

### ISSUE THREAD

**Issue by Satyavart** [macsnoeren/python-p2p-network#19]
It doesnt remove data of the disconnected node

**Comment by DoriHp** [comment]
Same problem when running nodes in multiple machines. Disconnect event didn't occurred.

**Comment by macsnoeren** [comment]
What is exactly the problem?

**Comment by Satyavart** [comment]
let's suppose there are 5 nodes naming 1 2 3 4 5.
such that it is connected in a way ............            1-->2-->3-->4-->5
If node 5 is sent offline,
     then node 4 won't get any message that node 5 is offline 
     it will still try to send a message to node 5
     and data of node 5 will still be saved in node 4 outbound_nodes.

**Comment by macsnoeren** [comment]
Okay. I understand that node 4 is not detecting that node 5 went offline. Is that correct? If so, yes I will going to solve that, while it should detect connection problems.

**Comment by Satyavart** [comment]
exactly
your outbound node disconnected & in bound node disconnected is not working properly

**Comment by macsnoeren** [comment]
Thanks, I will check it and propose a solution, I will come back here so you can check it ;).

**Comment by Satyavart** [comment]
yeah sure ..... if you want any help 
ping me at satyavarty591@gmail.com
I also have expertise in networks

**Comment by macsnoeren** [comment]
I have fixed the issue with the last commits to the development branch. Is it possible you could check it? The inbound and outbound data structures are correctly handled now and the events of disconnection is handled properly. Note that when a client has closed the connection, the other side will notice it after some time. I test this by sending multiple messages to detect the "broken pipe". If this occurs it closes the connection correctly.

**Comment by macsnoeren** [comment]
I also added a reconnect flag to the connect with node method. Whenever the connection is lost, the node will try to perform a reconnect.

**Comment by Satyavart** [comment]
I have sent you an mail regarding the problem and a possible solution.

**Comment by macsnoeren** [comment]
Do you mean that it does not work for you?

**Comment by macsnoeren** [comment]
I saw your e-mail and understand your solution. At this moment I implemented the solution, that when a broken connection is detected, everything is closed correctly. The proposed deauth message is in my opinion a solution, while this software is a framework. Everyone can implement this functionality if they want to.  The current code provide the solution. I will check more rubust code. At this moment I started to implement an Etheruem like blockchain with p2pnetwork. Thanks for your ideas and the issue.
